src/fields/field_python.impl.hh

```diff
@@ -109,18 +109,12 @@ void FieldPython<spacedim, Value>::set_func(const string &func_name)
     char func_char[func_name.size()+2];
     std::strcpy(func_char, func_name.c_str());
     p_func_ = PyObject_GetAttrString(p_module_, func_char );
-    if (! p_func_) {
-        if (PyErr_Occurred()) PyErr_Print();
-        xprintf(UsrErr, "Field '%s' not found in the python module: %s\n", func_name.c_str(), PyModule_GetName(p_module_) );
-        Py_XDECREF(p_func_);
-        Py_XDECREF(p_module_);
+    PythonLoader::check_error();
 
-    }
     if (! PyCallable_Check(p_func_)) {
-        xprintf(UsrErr, "Field '%s' from the python module: %s is not callable.\n", func_name.c_str(), PyModule_GetName(p_module_) );
-        Py_XDECREF(p_func_);
-        Py_XDECREF(p_module_);
-
+    	stringstream ss;
+    	ss << "Field '" << func_name << "' from the python module: " << PyModule_GetName(p_module_) << " is not callable." << endl;
+    	THROW(PythonLoader::ExcPythonError() << PythonLoader::EI_PythonMessage( ss.str() ));
```

Comment:
This error is not an exception raised in Python so it should not be reported through the PythonLoader exception. 
Move these checks into:
PyObject *PythonLoader::get_callable(PyObject *module, string name);
and raise dedicated exceptions defined in PythonLoader.